modules/tensor_mechanics/src/materials/ADComputeMultipleInelasticStress.C

```diff
@@ -318,11 +353,20 @@ ADComputeMultipleInelasticStress::computeAdmissibleState(
     ADRankTwoTensor & elastic_strain_increment,
     ADRankTwoTensor & inelastic_strain_increment)
 {
-  _models[model_number]->updateState(elastic_strain_increment,
-                                     inelastic_strain_increment,
-                                     _rotation_increment[_qp],
-                                     _stress[_qp],
-                                     _stress_old[_qp],
-                                     _elasticity_tensor[_qp],
-                                     _elastic_strain_old[_qp]);
+  if (_use_old_elasticity_tensor)
+    _models[model_number]->updateState(elastic_strain_increment,
+                                       inelastic_strain_increment,
+                                       _rotation_increment[_qp],
+                                       _stress[_qp],
+                                       _stress_old[_qp],
+                                       _elasticity_tensor_old[_qp],
+                                       _elastic_strain_old[_qp]);
+  else
+    _models[model_number]->updateState(elastic_strain_increment,
+                                       inelastic_strain_increment,
+                                       _rotation_increment[_qp],
+                                       _stress[_qp],
+                                       _stress_old[_qp],
+                                       _elasticity_tensor[_qp],
```

Comment:
wait... how does this compile then if `_elasticity_tensor[_qp]` is and ADR4T and `_elasticity_tensor_old[_qp]` is non-AD. Is there some overloading going on? Or is this implicitly upcasting the nonAD tensor to an AD tensor. That would be a bit wasteful.

Reply:
Yeah, not sure. Way easier this way then to make a new method.

Reply:
Plus it'd take some gymnastics because you cannot make function templates virtual. So, let's just leave it as is.